src/FakeItEasy/Core/ArgumentEqualityComparer.cs

```diff
@@ -57,39 +57,29 @@ public EnumerableComparer(Func<object?, object?, bool> elementComparer)
 
             public bool AreEqual(object expectedValue, object argumentValue)
             {
-                if (expectedValue is string expectedString && argumentValue is string argumentString)
-                {
-                    return string.Equals(expectedString, argumentValue);
-                }
-
                 if (argumentValue is not IEnumerable argumentEnumerable)
                 {
                     return false;
                 }
 
-                var expectedEnumerator = ((IEnumerable)expectedValue).GetEnumerator();
-                var actualEnumerator = argumentEnumerable.GetEnumerator();
-                var anyMoreExpected = expectedEnumerator.MoveNext();
-                var anyMoreActual = actualEnumerator.MoveNext();
-
-                while (anyMoreExpected && anyMoreActual)
-                {
-                    if (!this.areEqualElements(expectedEnumerator.Current, actualEnumerator.Current))
-                    {
-                        return false;
-                    }
-
-                    anyMoreExpected = expectedEnumerator.MoveNext();
-                    anyMoreActual = actualEnumerator.MoveNext();
-                }
-
-                if (anyMoreExpected ^ anyMoreActual)
+                if (expectedValue is string expectedString && argumentValue is string argumentString)
                 {
-                    return false;
+                    return string.Equals(expectedString, argumentString, StringComparison.Ordinal);
                 }
 
-                return true;
+                return Enumerable.SequenceEqual(
+                    ((IEnumerable)expectedValue).Cast<object?>(),
+                    argumentEnumerable.Cast<object?>(),
+                    this.elementComparer);

```

Comment:
nit: why not use `SequenceEqual` as an extension method?

Reply:
I didn't think of it! Doing so now.